### ISSUE TITLE
SFTP Repo?

### ISSUE THREAD

**Issue by flamingm0e** [ad-on-is/resticity#6]
Is there a way to add an SFTP repo?

I mostly run my backups to Wasabi, but I also have a storage VPS at HostHatch that I like to utilize (currently backing up my Nextcloud server there).  Currently running SFTPGO on that server as an endpoint for my NAS to back up to.

**Comment by ad-on-is** [comment]
Haven't tested yet, but you could mount the SFTP folder and use the local path option when adding the repo.

**Comment by flamingm0e** [comment]
Thanks.

I was hoping for native SFTP like restic offers.  I'll give it a shot though.

**Comment by ad-on-is** [comment]
Oh.. I totally did not see restic had SFTP support. I will add it.

**Comment by flamingm0e** [comment]
Oh you rock!  I was just setting up a docker container on my NAS to see if I could replace my Duplicacy backups.  That SFTP option would be tremendous there.

**Comment by ad-on-is** [comment]
1.0.30 is on it's way... Hopefully this should cover `sftp`, `rclone` and `rest` compatibility.

**Comment by flamingm0e** [comment]
You are awesome!

I'm so excited to have a GUI on my favorite backup utility

**Comment by ad-on-is** [comment]
Thank you 😅,

could you confirm that it works. I haven't had a chance to test it.

**Comment by flamingm0e** [comment]
I got it working!

I did have to init the repo first from the command line using an ssh config file due to port (I think Restic has a problem with this, so not your issue)

After figuring out the repo init issue, using `sftp:user@host:2022/restic` it connects and works fine.

Just a note: when adding a new repo the pre-filled text says smtp instead of sftp

https://github.com/ad-on-is/resticity/blob/5acae2c6deff6ec30a4e0b43eae3a953f88db865/frontend/components/Repository/New.vue#L72

**Comment by flamingm0e** [comment]
Trying to get this working in the Docker container and it throws an error about ssh not being available in the path, which I assume means the image needs open-ssh added?

**Comment by ad-on-is** [comment]
yes... restic indeed mentions to do all the necessary stuff in ~/.ssh/config.

Added openssh and rclone to the dockerfile and aur

**Comment by flamingm0e** [comment]
Yeah, I misread the Restic docs thinking it said if I needed to add a port TO an IPv6, not add a port OR IPv6.  :smile: I no read good

Thank you for all of this.  I will not have a chance to test the latest changes until tomorrow.

**Comment by ad-on-is** [comment]
All good. You've been of huge help already, finding this little things I missed :-)

**Comment by ad-on-is** [comment]
I'll close this one for now. If there's an issue feel free to reopen/open a new one.
